python/cudf/cudf/core/column/datetime.py

```diff
@@ -274,14 +276,68 @@ def as_numerical_column(
     def as_string_column(
         self, dtype: Dtype, format=None, **kwargs
     ) -> "cudf.core.column.StringColumn":
+        need_names_formats = {
```

Comment:
Can we move this dict out of this function and make a global dictionary at the top of this file? Something like..`DATETIME_SPECIAL_FORMATS`